src/include/rbd/librbd.hpp

```diff
@@ -532,6 +532,14 @@ class CEPH_RBD_API Image
   Image();
   ~Image();
 
+  // non-copyable
+  Image(const Image& rhs) = delete;
+  const Image& operator=(const Image& rhs) = delete;
```

Comment:
Probably doesn't actually matter here since it's a delete, but why is the returned reference has const?  Wouldn't a compiler-generated copy assignment operator return a non-const reference?

Reply:
It was const& in the original code. The canonical op= indeed returns a non-const reference. Better change it to non-const for consistency with move-assignment.